### ISSUE TITLE
Adding Catalan support

### ISSUE THREAD

**Issue by jmontane** [kamilmielnik/scrabble-solver#280]
Hi,
I'm the curator of an open licensed wordlist for Catalan language Scrabble. It's used at some Scrabble tournaments (other tournaments use a closed wordlist). I'm trying to add Catalan support to Scrabble-solver (See screnshots). But I have some issues:

- Multichar/digraph tiles. Catalan has NY, L·L and QU tiles. NY and QU works (I'm unable to type them, see Issue #279), but L·L tile (it's a 3 char tile) doesn't work. Neither using clipboard
- I can't provide dictionary definitions (how can avoid this requirement?)
- Catalan flag appears weird. I was unable to fix it
- I would like to classify Ç and L·L as consonant tiles, just like C, M or NY tiles.

Can you help me? Thanks in advance

![scrabble-solver-remaining-tiles](https://user-images.githubusercontent.com/339645/228331065-54fda0e4-21d3-49f1-abc7-dfb4dceab0bd.png)
![scrabble-solver-catalan](https://user-images.githubusercontent.com/339645/228331071-4686cc0b-4da2-45dd-9150-b3a33fc0778a.png)

**Comment by kamilmielnik** [comment]
> * Multichar/digraph tiles. Catalan has NY, L·L and QU tiles. NY and QU works (I'm unable to type them, see Issue [Unable to input Spanish digraph tiles #279](https://github.com/kamilmielnik/scrabble-solver/issues/279)), but L·L tile (it's a 3 char tile) doesn't work. Neither using clipboard

My fault, feel free to skip this and I'll make sure it's fixed before Catalan support is released.

>  (it's a 3 char tile) 

Ouuuuuch.
This not supported yet. It's hard but surely doable.
Feel free to skip this part.
I'll sleep on it and think about it.

> * I can't provide dictionary definitions (how can avoid this requirement?)

I believe we can use [Català Wiktionary](https://ca.wiktionary.org/wiki).
[It was used by Scrabble Solver at some point for all languages](https://github.com/kamilmielnik/scrabble-solver/pull/83/files) but unfortunately it was good-enough only for English.
If this is the best we can do for Catalan - let it be.
Feel free to skip this and I'll be more than happy to take care of this part.

> * Catalan flag appears weird. I was unable to fix it

Did you provide aspect ratio (see: [1](https://github.com/kamilmielnik/scrabble-solver/blob/master/packages/scrabble-solver/src/styles/variables.scss#L99-L105) & [2](https://github.com/kamilmielnik/scrabble-solver/blob/master/packages/scrabble-solver/src/i18n/i18n.module.scss))?

> * I would like to classify Ç and L·L as consonant tiles, just like C, M or NY tiles.

These constants have to be updated: https://github.com/kamilmielnik/scrabble-solver/blob/master/packages/constants/src/index.ts#L15-L17

I'm not sure about classifying L·L as a consonant. It's a digraph, right?

> Can you help me? Thanks in advance

It'll make me happy if Scrabble Solver supports Catalan :)

**Comment by kamilmielnik** [comment]
I've created a dedicated milestone for Catalan support: https://github.com/kamilmielnik/scrabble-solver/milestone/21

**Comment by jmontane** [comment]
> `L·L` is difficult because `·` is not allowed by itself. Would it be ok, if typing `LL` in Catalan would insert `L·L`?
> 
> The `·` would never have to be provided (but it would still work when pasting).

Yes, it's good enough. But two regular L can be played together too. See: _cella_ vs. _cel·la_ sample, they are different and valid words.

- C-E-L-L-A: 5 tiles are needed to play this word
- C-E-L·L-A: 4 tiles are needed to play this word

**Comment by kamilmielnik** [comment]
> I'll sleep on it and think about it.

I think it will be easier if your code is merged before I start [fixing the `L·L` case](https://github.com/kamilmielnik/scrabble-solver/issues/281) because I plan to do some refactoring that will create conflicts with your code.

> I can't provide dictionary definitions (how can avoid this requirement?)

I see in [your fork](https://github.com/jmontane/scrabble-solver) that you've added support for word definitions.
Can we close the #282 then?

**Comment by kamilmielnik** [comment]
> I think it will be easier if your code is merged before I start https://github.com/kamilmielnik/scrabble-solver/issues/281 because I plan to do some refactoring that will create conflicts with your code.

Not that many conflicts though. I'll proceed with it.

**Comment by kamilmielnik** [comment]
@jmontane FYI I've simplified adding languages a bit and wrote down a checklist for it: https://github.com/kamilmielnik/scrabble-solver#add-a-new-language. I hope the conflicts will not be that difficult to fix. I can handle them if they turn out to be too much of a trouble.

Would you mind submitting your PR?
From what I see it's pretty much done except some of the issues that are already reported separately in GitHub.
